File: đồ án thực tế CNPM/Server/src/main/java/com/example/hatd/model/dto/MatchNotificationDTO.java
package com.example.hatd.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
public class MatchNotificationDTO {

    private Long matchId;
    private String message; // Ví dụ: "Đã tìm thấy tài xế!"

    // Thông tin tài xế (gửi cho User)
    private String tenDriver;
    private String sdtDriver;
    private String bienSoXe;
    private String hangXe;

    private Double viDoDiemDi;
    private Double kinhDoDiemDi;
    private Double viDoDiemDen;
    private Double kinhDoDiemDen;

    // Thông tin khách (gửi cho Driver)
    private String tenUser;
    private String sdtUser;

    // Thông tin chuyến đi
    private String tenDiemDiUser;
    private String tenDiemDenUser;
    private Double giaTien;
    private LocalDateTime thoiGianDriverDenUser;
    private String hinhThucThanhToan;

    // Bạn có thể thêm bất cứ trường nào từ MatchedTrip mà UI cần
}

File: đồ án thực tế CNPM/Client/app/src/main/java/com/example/myhatd/data/model/MatchNotificationDTO.kt
package com.example.myhatd.data.model

// DTO này phải khớp với dữ liệu JSON bạn nhận từ Backend
data class MatchNotificationDTO(
    val matchId: Long?,
    val message: String?,

    // Thông tin tài xế
    val tenDriver: String?,
    val sdtDriver: String?,
    val bienSoXe: String?,
    val hangXe: String?,

    // Thông tin khách
    val tenUser: String?,
    val sdtUser: String?,

    // Thông tin chuyến đi
    val tenDiemDiUser: String?,
    val tenDiemDenUser: String?,
    val giaTien: Double?,
    val thoiGianDriverDenUser: String?,
    val hinhThucThanhToan: String?,

    val viDoDiemDi: Double?,
    val kinhDoDiemDi: Double?,
    val viDoDiemDen: Double?,
    val kinhDoDiemDen: Double?,
)